### ISSUE TITLE
FEATURE REQUEST - Prevent Screen Sleeping

### ISSUE THREAD

**Issue by Mind777** [clsid2/mpc-hc#1951]
Sometimes i pause a video and go away for some time, but when i return my monitor has switched off (went to sleep).

This can cause issues with resuming the video, and i usually have to exit the player and restart the video.

Can we please have an option to disable sleep/screensaver function so the monitor doesn't turn off when video is paused?

**Comment by clsid2** [comment]
Options > Advanced > ReloadAfterLongPause

**Comment by Mind777** [comment]
The problem with this option is that it can take 5-10 seconds to resume playback.

Also, I always use the "scale to 16:9 tv" option in mpc-hc to fill the entire screen. With the option you recommended, sometimes this resets and i have to re-enable it.

What would be great is to prevent the screen turning off when paused. So that playback can be resumed instantly and without error.

I am using madvr in fullscreen exclusive mode.

**Comment by clsid2** [comment]
You can prevent screen from turning off in the Windows power settings.

I am NOT going to add such options in MPC for paused state.

**Comment by Mind777** [comment]
My computer is multi-purpose use throughout the day, with both a monitor and tv attached. So toggling the windows power setting on/off would be a real pain. This really needs to be controlled at the application level. 

A quick search shows similiar functionality has been requested since 2014, so i guess if it's not been implemented now it never will be. Hence the "feature request".

The "reload after long pause" is a nice idea, but practically doesn't work well. At least in my case with madvr.

For anyone with this same issue, i found Process Lasso has an option to prevent the screen switching off/sleeping when mpc-hc is running, regardless of if video is paused. Just a shame to have to install an extra program for such a simple function.

**Comment by ale5000-git** [comment]
The fact that the `ReloadAfterLongPause` option is needed is because of buggy drivers, on many PC it works without enabling this option.

**Comment by Mind777** [comment]
> The fact that the `ReloadAfterLongPause` option is needed is because of buggy drivers, on many PC it works without enabling this option.

If the reload from pause isn't too long (say after a few minutes), it resumes without issue. The problem is for longer pauses, then I get the resume issue occuring (like 10min+). 

Running windows 11 with latest nvidia drivers.

**Comment by clsid2** [comment]
You can also try the "Reinitialize when changing display" option for the EVR-CP video renderer.

Or simply pressing left arrow to seek backwards a tiny bit before resuming playback may also help.

Ultimately it is a driver bug and having the latest one is no guarantee at all that it is bug free.

**Comment by Mind777** [comment]
> You can also try the "Reinitialize when changing display" option for the EVR-CP video renderer.

I'm using MadVR so I don't have that option. I've tried both exclusive and windowed mode and same issue.

Was watching a movie last night and paused halfway through, when i came back the "scale to 16:9 tv" had reset and picture was letterboxed. The audio also went out of sync, so i had to exit the player and reload.

I've disabled "ReloadAfterLongPause" now as it seems to mess up playback for me after a long pause unfortunately.
